src/debug.cc

```diff
@@ -902,8 +903,10 @@ Raw::print(std::ostream &os) const
         if (data_) {
             if (useHex_)
                 printHex(os);
-            else
-                os.write(data_, size_);
+            else {
+                auto len = (limit_ > 0 && limit_ < size_ ? limit_ : size_);
+                os.write(data_, len);
+            }
```

Comment:
Please provide a visual clue to the reader that the data was truncated:

```suggestion
        if (printableSize() < size_)
            os << "[...]";
```

You can even print the number of truncated bytes: `[... N more bytes ...]`.

Reply:
Done in 33cdf16